components/Input/Input.tsx

```diff
@@ -0,0 +1,439 @@
+        px: '$2',
+        lineHeight: '$sizes$5',
+        '&:-webkit-autofill::first-line,&:autofill::first-line': {
+          fontSize: '$1',
+        },
+      },
+      medium: {
+        height: MEDIUM_HEIGHT,
+        fontSize: '$3',
+        px: '$3',
+        lineHeight: '$sizes$6',
+        '&:-webkit-autofill::first-line,&:autofill::first-line': {
+          fontSize: '$3',
+        },
+      },
+      large: {
+        height: LARGE_HEIGHT,
+        fontSize: '$3',
+        px: '$3',
+        lineHeight: '$sizes$7',
+        '&:-webkit-autofill::first-line,&:autofill::first-line': {
+          fontSize: '$3',
+        },
+      },
+    },
+    startAdornment: {
+      true: {},
+    },
+    endAdornment: {
+      true: {},
+    },
+    variant: {
+      ghost: {
+        boxShadow: 'none',
+        '&:disabled': {
+          boxShadow: 'none',
+        },
+        backgroundColor: 'transparent',
+        '&:focus-visible': {
+          backgroundColor: '$loContrast',
+        },
+      },
+    },
+    state: {
+      invalid: {
+        boxShadow: 'inset 0 0 0 1px $colors$inputInvalidBorder',
+      },
+    },
+    cursor: {
+      default: {
+        cursor: 'default',
+        '@hover': {
+          '&:hover:not(:read-only)': {
+            cursor: 'text',
+          },
+        },
+        '&:focus-visible:not(:read-only)': {
+          cursor: 'text',
+        },
+      },
+      text: {
+        cursor: 'text',
+      },
+    },
+  },
+  defaultVariants: {
+    size: 'medium',
+    cursor: 'default',
+  },
+  compoundVariants: [
+    {
+      startAdornment: true,
+      size: 'small',
+      css: {
+        paddingInlineStart: 'calc($2 + 15px)',
+      },
+    },
+    {
+      startAdornment: true,
+      size: 'medium',
+      css: {
+        paddingInlineStart: 'calc($3 + 15px)',
+      },
+    },
+    {
+      startAdornment: true,
+      size: 'large',
+      css: {
+        paddingInlineStart: 'calc($3 + 15px)',
+      },
+    },
+    {
+      endAdornment: true,
+      size: 'small',
+      css: {
+        paddingInlineEnd: 'calc($2 + 15px)',
+      },
+    },
+    {
+      endAdornment: true,
+      size: 'medium',
+      css: {
+        paddingInlineEnd: 'calc($3 + 15px)',
+      },
+    },
+    {
+      endAdornment: true,
+      size: 'large',
+      css: {
+        paddingInlineEnd: 'calc($3 + 15px)',
+      },
+    },
+  ],
+});
+
+const InputWrapper = styled('div', {
+  // Reset
+  outline: 'none',
+  lineHeight: 0,
+
+  position: 'relative',
+  backgroundColor: '$inputBg',
+  color: '$inputPlaceholder',
+
+  '&::before': {
+    boxSizing: 'border-box',
+    content: '""',
+    position: 'absolute',
+    top: 0,
+    right: 0,
+    bottom: 0,
+    left: 0,
+  },
+  '&::after': {
+    boxSizing: 'border-box',
+    content: '""',
+    position: 'absolute',
+    top: 0,
+    right: 0,
+    bottom: 0,
+    left: 0,
+  },
+
+  '&:focus-visible': {
+    '&::before': {
+      backgroundColor: '$inputFocusBg',
```

Comment:
As far as I understand, you try to apply this value as a shadow, but it is either applied to the whole background - when I set `focus-visible` on the dom node - or not at all.

`StyledInput` now has a transparent background. I think the result is not what you expected.